os/kernel/log_dump/log_dump.h

```diff
@@ -0,0 +1,46 @@
+/****************************************************************************
+ *
+ * Copyright 2022 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/mm/mm.h>
+#include <queue.h>
+
+/********************************************************************************
+ * Definitions
+ ********************************************************************************/
+
+#define LOG_DUMP_ERROR		-1
+#define LOG_DUMP_SUCCESS	 0
+
+/********************************************************************************
+ * Public Types
+ ********************************************************************************/
+
+struct log_dump_chunk_s {
+	struct log_dump_chunk_s *flink;
+	char arr[CONFIG_LOG_DUMP_CHUNK_SIZE];
+};
+
+/********************************************************************************
+ * Public Functions
+ ********************************************************************************/
+
+int log_dump_init(void);
```

Comment:
Let's split internal header and external header.
`int log_dump_init(void);` is for external. Others are for internal.